testsuite/tests/typing-jkind-bounds/basics.ml

```diff
@@ -1589,14 +1589,5 @@ class type sexp_of =
 
 [%%expect{|
 type t = Atom of string | List of t list
-Line 1:
-Error: Class type declarations do not match:
-         class type sexp_of =
-           object method array : ('a -> t) -> 'a array -> t end
-       does not match
-         class type sexp_of =
-           object method array : ('a -> t) -> 'a array -> t end
-       The method array has type "'a. ('a -> t) -> 'a array -> t"
-       but is expected to have type "'a. ('a -> t) -> 'a array -> t"
-       Type "'a" is not compatible with type "'a0"
+class type sexp_of = object method array : ('a -> t) -> 'a array -> t end
```

Comment:
The change in the commit is fine, but do we understand how it fixes the problem? I want to be sure we are missing another place where it should be resetting `univar_pairs`, or nested calls to functions that aren’t reentrant.

Reply:
I went through all uses of `univar_pairs` to use the protect-refs approach on all of them, not just the specific places that caused the bug. I sadly don't now remember which specific one it was. I suppose I could remove the fix and re-debug if you wanted more information.